### PR TITLE
Add `pp_error` to FLOW and use a result type

### PR DESCRIPTION
**Pull request by samoht**
Related to #381

### CODE REVIEW

File: types/V1.mli [hunk 1]
```diff
@@ -29,6 +29,9 @@ module type DEVICE = sig
   type error
   (** An error signalled by the device *)
 
+  val pp_error: Format.formatter -> error -> string
```
Comment (hannesm) [inline types/V1.mli]: this should now be `unit` here!?

Comment (samoht) [inline types/V1.mli]: it was just to check if someone was following :-)

**Comment by samoht** [comment]
These patches should be revisited now that we have a `Rresult` and `Fmt`. I let that issue open to remember (until we got a Mirage's Cuekeeper instance to manage the project)

**Comment by hannesm** [comment]
I'm very much in favour of reviving this PR, and since this breaks the API, we should also rename `V1` and `V1_LWT` to `M` and `M_LWT` (or some other name which does not include versioning information).  Versioning is handled by opam :)

**Comment by yomimono** [comment]
I'm closing this PR as we probably won't incorporate these exact commits, and we've merged quite a few other things that conflict with it.  But the spirit lives on in https://github.com/mirage/mirage-dev/pull/166 and associated commits and discussions, and we'll likely end up merging something resembling this.  Thanks very much, @samoht, for leading the charge here!

**Pull request closed without merging**